mvp_example/src/main/java/main/MvpModule.kt

```
@@ -0,0 +1,30 @@
+package main
+
+import android.util.Log
+import androidx.lifecycle.LifecycleOwner
+import com.distillery.android.blueprints.mvp.todo.contract.TodoContract
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import org.koin.core.qualifier.named
+import main.presenter.PresenterImplementation
+import org.koin.dsl.module
+
+@ExperimentalCoroutinesApi
+val mvpModule = module {
+    factory(named(TAG)) {
+        val job = Job()
+        val errorHandler = CoroutineExceptionHandler { _, throwable ->
```

Using error handler to log errors is fine, but I think we should rethrow them, they are meant to crash the app, so if they got here - let it crash, track it and then fix the root cause, otherwise it will mask real issues

---

Ok, so to be sure the idea is if we reach this point let throw the exception and let it crash correct?

---

yeap